docs/src/opensourcetpa.md

```diff
@@ -0,0 +1,294 @@
+
+```
+sudo update-grub
+sudo reboot
+```
+
+**WARNING**: Giving a user the ability to speak to the Docker daemon
+lets them trivially gain root on the Docker host. Only trusted users
+should have access to the Docker daemon.
+
+### Creating a configuration with TPA
+
+The next step in this process is to create a configuration. TPA does most of the work for you through its `configure` command. All you have to do is supply command line flags and options to select, in broad terms, what you want to deploy. Here's our `tpaexec configure` command:
+
+```
+tpaexec configure demo --architecture M1 --platform docker --postgresql 15 --enable-repmgr --no-git
+```
+
+This creates a configuration called `demo` which has the [M1 architecture](architecture-M1/). It will therefore have a primary, replica and backup node.
+
+The `--platform docker` tells TPA that this configuration should be created on a local Docker instance; it will provision all the containers and OS requirements. Other platforms include [AWS](platform-aws), which does the same with Amazon Web Services and [Bare](platform-bare), which skips to operating system provisioning and goes straight to installing software on already configured Linux hosts.
+
+With `--postgresql 15`, we instruct TPA to use Community Postgres, version 15. There are several options here in terms of selecting software, but this is the most straightforward default for open-source users.
+
+Adding `--enable-repmgr` tells TPA to use configure the deployment to use [Replication Manager](https://www.repmgr.org/) to hand replication and failover. 
+
+Finally, `--no-git` turns off the feature in TPA which allows you to revision control your configuration through git. 
+
+Run this command, and apparently, nothing will happen on the command line. But you will find a directory called `demo` has been created containing some files including a `config.yml` file which is a blueprint for our new deployment.
+
+## Provisioning the deployment
+
+Now we are ready to create the containers (or virtual machines) on which we will run our new deployment. This can be achieved with the `provision` command. Run:
+
+```
+tpaexec provision demo
+```
+
+You will see TPA work through the various operations needed to prepare for deployment of your configuration. 
+
+## Deploying
+
+Once provisioned, you can move on to deployment. The installs, if needed, operating systems and system packages. It then installs the requested Postgres architecture and performs all the needed configuration. 
+
+```
+tpaexec deploy demo
+```
+
+You will see TPA work through the various operations needed to deploy your configuration.
+
+
+## Testing
+
+You can quickly test your newly deployed configuration using the tpaexec `test` command which will run pgbench on your new database.
+
+```
+tpaexec test demo
+```
+
+## Connecting
+
+To get to a psql prompt, the simplest route is to log into one of the containers (or VMs or host depending on configuration) using SSH. Run 
```

Comment:
I find the simplest route is `docker exec -it hostname /bin/bash` any reason to use SSH beyond documenting these potentially useful commands?

Reply:
We're using Docker in this guide though, but I see your point. Perhaps we should just note my method as a shortcut for people with no interest in beyond Docker.

Reply:
Adding in a wrapper to cover both.